src/Altinn.AccessManagement.Core/Services/AppsInstanceDelegationService.cs

```diff
@@ -0,0 +1,292 @@
+
+            try
+            {
+                delegableRights = await _pip.GetDelegableRightsByApp(resourceQuery, cancellationToken);
+            }
+            catch (ValidationException)
+            {
+                errors.Add(ValidationErrors.MissingPolicy, "appInstanceDelegationRequest.Resource");
+            }
+
+            if (delegableRights == null || delegableRights.Count == 0)
+            {
+                errors.Add(ValidationErrors.MissingDelegableRights, "appInstanceDelegationRequest.Resource");
+            }
+        }
+
+        if (errors.TryBuild(out var errorResult))
+        {
+            return errorResult;
+        }
+
+        result.From = appsInstanceDelegationRequest.From;
+        result.To = appsInstanceDelegationRequest.To;
+        result.ResourceId = appsInstanceDelegationRequest.ResourceId;
+        result.InstanceId = appsInstanceDelegationRequest.InstanceId;
+        result.InstanceDelegationMode = appsInstanceDelegationRequest.InstanceDelegationMode;
+
+        // Perform delegation
+        DelegationHelper.TryGetPerformerFromAttributeMatches(appsInstanceDelegationRequest.PerformedBy, out string performedById, out UuidType performedByType);
+        InstanceRight rulesToDelegate = new InstanceRight
+        {
+            FromUuid = (Guid)from.Uuid,
+            FromType = from.Type,
+            ToUuid = (Guid)to.Uuid,
+            ToType = to.Type,
+            PerformedBy = performedById,
+            PerformedByType = performedByType,
+            ResourceId = appsInstanceDelegationRequest.ResourceId,
+            InstanceId = appsInstanceDelegationRequest.InstanceId,
+            InstanceDelegationMode = appsInstanceDelegationRequest.InstanceDelegationMode
+        };
+        List<RightInternal> rightsAppCantDelegate = new List<RightInternal>();
+        List<RightV2DelegationResult> rights = new List<RightV2DelegationResult>();
+        UrnJsonTypeValue instanceId = KeyValueUrn.CreateUnchecked($"{AltinnXacmlConstants.MatchAttributeIdentifiers.ResourceInstanceAttribute}:{appsInstanceDelegationRequest.InstanceId}", AltinnXacmlConstants.MatchAttributeIdentifiers.ResourceInstanceAttribute.Length + 1);
+        
+        foreach (RightInternal rightToDelegate in appsInstanceDelegationRequest.Rights)
+        {
+            if (CheckIfInstanceIsDelegable(delegableRights, rightToDelegate))
+            {
+                rightToDelegate.Resource.Add(instanceId);
+                rulesToDelegate.InstanceRules.Add(new InstanceRule
+                {
+                    Resource = rightToDelegate.Resource,
+                    Action = rightToDelegate.Action
+                });
+            }
+            else
+            {
+                rightsAppCantDelegate.Add(rightToDelegate);
+            }
+        }
+
+        InstanceRight delegationResult = await _pap.TryWriteInstanceDelegationPolicyRules(rulesToDelegate, cancellationToken);
```

Comment:
_pap.TryWriteInstanceDelegationPolicyRules is called even when rulesToDelegate is empty causing policy with empty target to be written to storage and change to database